klusterletconfig/v1alpha1/types.go

```diff
@@ -124,6 +139,88 @@ type Registries struct {
 	Source string `json:"source"`
 }
 
+// KubeAPIServerConfig specifies the custom configuration for the Hub kube API server
+type KubeAPIServerConfig struct {
+	// URL is the endpoint of the hub Kube API server.
+	// If not present, the .status.apiServerURL of Infrastructure/cluster will be used as the default value.
+	// e.g. `oc get infrastructure cluster -o jsonpath='{.status.apiServerURL}'`
+	// +optional
+	URL string `json:"url,omitempty"`
+
+	// ServerVerificationStrategy is the strategy used for verifying the server certification
+	// +kubebuilder:default=default
+	// +kubebuilder:validation:Enum=use-system-truststore;use-auto-detected-truststore;use-custom-ca-bundles;default
+	// +optional
+	ServerVerificationStrategy ServerVerificationStrategy `json:"serverVerificationStrategy,omitempty"`
+
+	// TrustedCABundles refers to a collection of user-provided CA bundles used for verifying the server
+	// certificate of the hub Kubernetes API
+	// If the ServerVerificationStrategy is set to "use-system-truststore", this field will be ignored.
+	// Otherwise, the CA certificates from the configured bundles will be appended to the klusterlet CA bundle.
+	// +listType:=map
+	// +listMapKey:=name
+	// +optional
+	TrustedCABundles []CABundle `json:"trustedCABundles,omitempty"`
+
+	// ProxyURL is the URL to the proxy to be used for all requests made by client
+	// If an HTTPS proxy server is configured, you may also need to add the necessary CA certificates to
+	// TrustedCABundles.
+	// +optional
+	ProxyURL string `json:"proxyURL,omitempty"`
+}
+
+// CABundle is a user-provided CA bundle
+type CABundle struct {
+	// Name is the identifier used to reference the CA bundle
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name,omitempty"`
+
+	// CABundleData is base64 encoded certificate data
+	// +optional
+	CABundleData []byte `json:"caBundleData,omitempty"`
+
+	// CABundle refers to a ConfigMap containing the user-provided CA bundle
+	// The key of the CA data must be "ca-bundle.crt".
```

Comment:
We may have to accept multiple keys in the configmap, including `ca-bundle.crt`, ` ca.crt` and `tls.crt`

Reply:
We might also need a label to filter the ConfigMaps when monitoring the ConfigMap resource.